tendrl/node_agent/manager/pull_hardware_inventory.py

```diff
@@ -122,6 +126,105 @@ def getTendrlContext():
     return tendrl_context
 
 
+def get_node_disks():
+    '''returns structure
+
+    {[{"device_name":   "devicename",
+       "fs_type":     "fstype",
+       "serial_no": disk['SERIAL'],
+       "firmware_version": disk['FWVERSION'],
+       "fs_uuid":     "uuid",
+       "model":      "model",
+       "mount_point": ["mountpoint", ...],
+       "name":       "name",
+       "parent":     "parentdevicename",
+       "size":       uint64,
+       "disk_type":       "type",
+       "used":       boolean,
+       "ssd":        boolean,
+       "vendor":     "string",
+       "disk_id":     "uuid"}, ...], ...}
+    '''
+    rv = []
+    columns = 'NAME,KNAME,FSTYPE,MOUNTPOINT,UUID,PARTUUID,MODEL,SIZE,TYPE,' \
+              'PKNAME,VENDOR,ROTA'
+    keys = columns.split(',')
+    lsblk = ("lsblk --all --bytes --noheadings --output='%s' --path --raw" %
+             columns)
+    cmd = Command({"_raw_params": lsblk})
+    out, err = cmd.start()
+    if not out['stderr']:
+        devlist = {}
+        dev_info = {}
+        devlist = map(lambda line: dict(zip(keys, line.split(' '))),
+                      out['stdout'].splitlines())
+        parents = set([d['PKNAME'] for d in devlist if 'PKNAME' in d])
+        for d in devlist:
+            in_use = True
+            if d['TYPE'] not in ['disk', 'part']:
+                continue
+            if d['TYPE'] == 'disk':
+                if d['KNAME'] in parents:
+                    # skip it
+                    continue
+                else:
+                    in_use = False
+            elif not d['FSTYPE']:
+                in_use = False
+
+            d.update({'INUSE': in_use})
+            dev_info.update({d['KNAME']: d})
+            cmd = ("hdparm -I %s" % d['KNAME'])
+            cmd = Command({"_raw_params": cmd})
+            out, err = cmd.start()
+            if not out['stderr']:
+                out = out['stdout']
+                info_list = out.split('\n')
+                d['SERIAL'] = info_list[5].split(':')[1].strip()
```

Comment:
This is extremely fragile. It relies on hdparm outputting a specific information in a very specific sequence. At least parse for a specific string and then extract the information. Even in that case, it is still fragile, but nowhere near as bad as this. And don't use an exact string match, just use a string that's specific enough to give you the line that contains the information you want.